src/gpu/symmetry.jl

```diff
@@ -0,0 +1,46 @@
+using GPUArraysCore
+
+function accumulate_over_symmetries!(ρaccu::AbstractArray, ρin::AbstractArray,
+    basis::PlaneWaveBasis{T}, symmetries) where {T}
+    Gs = reshape(G_vectors(basis), size(ρaccu))
+    fft_size = basis.fft_size
+
+    symm_invS = to_device(basis.architecture, [Mat3{Int}(inv(symop.S)) for symop in symmetries])
+    symm_τ = to_device(basis.architecture, [symop.τ for symop in symmetries])
+    n_symm = length(symmetries)
+
+    map!(ρaccu, Gs) do G
+        acc = zero(complex(T))
+        # Explicit loop over indicies because AMDGPU does not support zip() in map!
+        for i_symm in 1:n_symm
+            invS = symm_invS[i_symm]
+            τ = symm_τ[i_symm]
+            idx = index_G_vectors(fft_size, invS * G)
+            acc += isnothing(idx) ? zero(complex(T)) : cis2pi(-T(dot(G, τ))) * ρin[idx]
+        end
+        acc
+    end
+    ρaccu
+end
+
+function lowpass_for_symmetry!(ρ::AbstractGPUArray, basis::PlaneWaveBasis{T};
```

Comment:
The only place where we need this is in `symmetrize_ρ`, where it comes right after `accumulate_over_symmetries!`. I think for the GPU version it would make a lot of sense to fuse these two functions into one and thus have a single `map!` going over all `Gs`. This you should be able to do with a boolean flag `do_lowpass`, which hopefully Julia is smart enough to constant-prop into the GPU kernel and fully compile away if set to `false`.

Again feel free to make this fused function also the CPU function if this does not hurt performance too much.